### PR TITLE
Release CLI v12

### PR DESCRIPTION
**Pull request by pdcastro**
Resolves: #1770
Change-type: major

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,7 +56,10 @@
     "package": "npm run build:fast && npm run build:standalone && npm run build:installer",
     "release": "ts-node --transpile-only automation/run.ts release",
     "pretest": "npm run build",
-    "test": "mocha --timeout 6000 -r ts-node/register/transpile-only \"tests/**/*.spec.ts\"",
+    "test": "npm run test:source && npm run test:standalone",
+    "test:source": "mocha --timeout 6000 -r ts-node/register/transpile-only \"tests/**/*.spec.ts\"",
+    "test:standalone": "npm run build:standalone && cross-env BALENA_CLI_TEST_TYPE=standalone npm run test:source",
+    "test:standalone:fast": "cross-env BALENA_CLI_TEST_TYPE=standalone npm run test:source",
```
Comment (pdcastro) [inline package.json:62]: I should highlight these changes for reviewing purposes. The are related to enabling the tests to run with the standalone executable. Btw, see comments at the top of `tests/proxy-server.ts` for an overview of the tests involving a proxy server.

File: tests/proxy-server.ts [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+ *    child process to use HTTP instead of HTTPS for the balena API and builder.)
+ * 4. The child process sends the HTTP requests to the proxy server.
+ * 5. The proxy server forwards the request to the interceptor server.
+ * 6. The interceptor server simply re-issues the HTTP request (unchange), with
+ *    the expectation that `nock` will intercept it.
+ * 7. `nock` (running on the parent process, same process that runs `mocha`)
+ *    intercepts the HTTP request, test it and replies with a mocked response.
+ * 8. `nocks` response is returned to the interceptor server, which returns it
+ *    to the proxy server, which returns it to the child process, which continues
+ *    CLI command execution.
```
Comment (pdcastro) [inline tests/proxy-server.ts:54]: Check these comments for an explanation of the standalone executable tests involving a proxy server.

File: tests/supervisor-mock.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+			'{"message":"Streaming logs","isSystem":true}\n',
+			'{"serviceName":"bar","serviceId":1,"imageId":1,"isStdout":true,"timestamp":1591991625223,"message":"bar 8 (332) Linux 4e3f81149d71 4.19.75 #1 SMP PREEMPT Mon Mar 23 11:50:49 UTC 2020 aarch64 GNU/Linux"}\n',
+			'{"serviceName":"foo","serviceId":2,"imageId":2,"isStdout":true,"timestamp":1591991628757,"message":"foo 8 (200) Linux cc5df60d89ee 4.19.75 #1 SMP PREEMPT Mon Mar 23 11:50:49 UTC 2020 aarch64 GNU/Linux"}\n',
+		];
+		let chunkCount = 0;
+		const chunkedStream = new Readable({
+			read(_size) {
+				setTimeout(() => {
+					this.push(chunkCount === chunks.length ? null : chunks[chunkCount++]);
+				}, 10);
```
Comment (pdcastro) [inline tests/supervisor-mock.ts:49]: This `chunkedStream` mimics the supervisor behavior on a device, incrementally producing JSON chunks on a keepalive TCP connection (btw, not HTTP keepalive, TCP keepalive... the HTTP header is `connection: close`, as I learned the hard way while writing these tests) :-)

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -224,4 +224,3 @@ gotchas to bear in mind:
   `node_modules/balena-sdk/node_modules/balena-errors`  
   In the case of subclasses of `TypedError`, a string comparison may be used instead:  
   `error.name === 'BalenaApplicationNotFound'`
-
```
Comment (pdcastro) [inline CONTRIBUTING.md]: This commit is practically empty (just this blank line deletion), and used to highlight the Release Notes URL on the commit message, at the top of the list of the v12.0.0 commits in the CLI changelog.

File: package.json (hunk 1 above)
Comment (srlowe) [reply package.json:62]: Nice!

**Pull request merged**